server/ServerNetworking.cpp

```diff
@@ -207,7 +207,17 @@ void PlayerConnection::SendMessage(const Message& message) {
         ErrorLogger(network) << "PlayerConnection::SendMessage can't send message when not transmit connected";
         return;
     }
-    m_service.post(boost::bind(&PlayerConnection::SendMessageImpl, shared_from_this(), message));
+    m_service.post(boost::bind(&PlayerConnection::SendMessageImpl, shared_from_this(), message, std::function<void(bool)>()));
+}
+
+void PlayerConnection::SendMessage(const Message& message, std::function<void(bool)> callback) {
```

Comment:
Could maybe just return a bool from thus and avoid the callback stuff?

Reply:
Sending will be completed asynchronously later in `PlayerConnection::HandleMessageWrite`. I sometimes get notes from players they didn't get notifications because the server treats them as connected before sending next turn message and then disconnects them on sending error.